binr/r2r/r2r.c

```diff
@@ -464,6 +464,7 @@ int main(int argc, char **argv) {
 		RThread *th = r_th_new (worker_th, &state, 0);
 		if (!th) {
 			eprintf ("Failed to start thread.\n");
+			r_th_lock_leave (state.lock);
 			exit (-1);
```

Comment:
exit should free all resources so it shouldnt be a problem imho. but it's formally correct to do so i guess :)

Reply:
Hi, indeed, it is harmless since most of the OS(some OS do not like embedded systems) would release the resources when exiting. I think releasing a lock at the source code level is good for resource management : )